spec/models/place_spec.rb

```diff
@@ -56,10 +56,24 @@
   end
 
   describe "#image_location" do
-    it "concatenates the image host with the image_uri" do
-      place = build(:place, image_uri: "fake_image_code.jpg")
+    context "when there is an image hostname env var" do
+      it "concatenates the image hostname with the image_uri" do
+        ClimateControl.modify IMAGE_HOSTNAME: "http://test" do
+          place = build(:place, image_uri: "fake_image_code.jpg")
 
-      expect(place.image_location).to eql("test/fake_image_code.jpg")
+          expect(place.image_location).to eql("http://test/fake_image_code.jpg")
+        end
+      end
+    end
+
+    context "when there is no image hostname env var" do
+      it "concatenates the geograph_photos folder name with the image_uri" do
+        ClimateControl.modify IMAGE_HOSTNAME: nil do
+          place = build(:place, image_uri: "fake_image_code.jpg")
+
+          expect(place.image_location).to eql("geograph_photos/fake_image_code.jpg")
```

Comment:
This is my first time looking at this, and the changes in the PR look good! But my spidey sense is tingling about what's happening elsewhere here to ambiguously passing URLs and local filenames around. Is any of this user configurable? Is there a possible attack where someone makes a filename something like `../../../../../etc/root/passwords` and exposes files on the system?

Reply:
Users can't upload images to the live site - all the images have been downloaded from Geograph, but DataSciLab controls if any new images are added.